### PR TITLE
Fix data race of proxy ut

### PR DESCRIPTION
**Pull request by longjiquan**
Signed-off-by: dragondriver <jiquan.long@zilliz.com>
issue: #8083

### CODE REVIEW

**Review by xiaofan-luan — APPROVED** [review]

File: internal/distributed/indexnode/client/client.go [hunk 1]
```diff
@@ -37,12 +37,43 @@ type Client struct {
...
+	}
+
+	return c.grpcClient, nil
+}
+
+func (c *Client) resetConnection() {
+	c.grpcClientMtx.Lock()
+	defer c.grpcClientMtx.Unlock()
+
+	c.conn = nil
```
Comment (xiaofan-luan) [inline internal/distributed/indexnode/client/client.go:73]: should close the conn when reset

Comment (DragonDriver) [reply internal/distributed/indexnode/client/client.go:73]: OK

**Pull request merged**